Apps/Common/src/Delegates/PHSA/RestVaccineStatusDelegate.cs

```diff
@@ -194,6 +195,53 @@ public async Task<RequestResult<PHSAResult<VaccineStatusResult>>> GetVaccineStat
             return retVal;
         }
 
+        /// <inheritdoc/>
+        public async Task<RequestResult<PHSAResult<VaccineStatusResult>>> GetVaccineStatusWithRetries(VaccineStatusQuery query, string accessToken, bool isPublicEndpoint)
+        {
+            using Activity? activity = Source.StartActivity("RetryGetVaccineStatus");
+            RequestResult<PHSAResult<VaccineStatusResult>> retVal = new ()
+            {
+                ResultStatus = ResultType.Error,
+                PageIndex = 0,
+            };
+
+            RequestResult<PHSAResult<VaccineStatusResult>> response;
+            int attemptCount = 0;
+            bool refreshInProgress;
+            do
+            {
+                response = await this.GetVaccineStatus(query, accessToken, isPublicEndpoint).ConfigureAwait(true);
+
+                refreshInProgress = response.ResultStatus == ResultType.Success &&
+                                    response.ResourcePayload != null &&
+                                    response.ResourcePayload.LoadState.RefreshInProgress;
+
+                attemptCount++;
+                if (refreshInProgress && attemptCount <= this.phsaConfig.MaxRetries)
+                {
+                    this.logger.LogDebug($"Refresh in progress, trying again....");
+                    Thread.Sleep(this.phsaConfig.BackOffMilliseconds);
```

Comment:
I believe we want this to be our phsa config backoff or the returned backoff whichever is greater.